cmd/precise-code-intel-bundle-manager/internal/database/cache_test.go

```diff
@@ -3,28 +3,70 @@ package database
 import (
 	"errors"
 	"fmt"
-	"strings"
+	"sync"
 	"testing"
 	"time"
 )
 
 func TestDatabaseCacheEvictionWhileHeld(t *testing.T) {
+	// keep track of what db mocks are closed
+	closed := map[Database]bool{}
+	// protected concurrent access to closed map
+	var mutex sync.Mutex
+
+	// openTestDatabase creates a new Database that inserts an entry
+	// for itself into the close map when its Close method is called.
+	openTestDatabase := func() (Database, error) {
+		db := NewMockDatabase()
+		db.CloseFunc.SetDefaultHook(func() error {
+			mutex.Lock()
+			defer mutex.Unlock()
+			closed[db] = true
+			return nil
+		})
+
+		return db, nil
+	}
+
+	// isOpen returns true if the database has not yet been closed.
+	isOpen := func(db Database) bool {
+		mutex.Lock()
+		defer mutex.Unlock()
+		return !closed[db]
+	}
+
+	// isOpenFor100ms will call isOpen for the given database until it
+	// has closed or 250ms has elapsed. This is used to test whether or
+	// not a database handle has been closed by an LRU eviction.
+	isOpenFor250ms := func(db Database) bool {
```

Comment:
Looking at the function body I think it is 100ms?
```suggestion
	// isOpenFor100ms will call isOpen for the given database until it
	// has closed or 100ms has elapsed. This is used to test whether or
	// not a database handle has been closed by an LRU eviction.
	isOpenFor100ms := func(db Database) bool {
```

What about accept another argument (e.g. `n`) to indicate how many ms to wait?